src/ResourceManager/Compute/Commands.Compute/Common/Utils.cs

```diff
@@ -0,0 +1,151 @@
+
+                      "Standard_D14_v2", "Standard_D5_v2_Promo", "Standard_D14_v2_Promo", "Standard_DS5_v2",
+
+                      "Standard_DS14_v2", "Standard_DS5_v2_Promo", "Standard_DS14_v2_Promo", "Standard_F16",
+
+                      "Standard_F16s", "Standard_M64-32ms", "Standard_M128-32ms", "Standard_D32_v3",
+
+                      "Standard_D32s_v3", "Standard_D64-32s_v3", "Standard_E32_v3", "Standard_E32s_v3",
+
+                      "Standard_E32-8s_v3", "Standard_E32-16_v3", "Standard_D5_v2_ABC", "Standard_D14_v2_ABC",
+
+                      "Standard_F16_ABC", "Standard_F32s_v2", "Standard_D15_v2", "Standard_D15_v2_Promo",
+
+                      "Standard_D15_v2_Nested", "Standard_DS15_v2", "Standard_DS15_v2_Promo",
+
+                      "Standard_DS15_v2_Nested", "Standard_D40_v3", "Standard_D40s_v3", "Standard_D15_v2_ABC",
+
+                      "Standard_M64ms", "Standard_M64s", "Standard_M128-64ms", "Standard_D64_v3", "Standard_D64s_v3",
+
+                      "Standard_E64_v3", "Standard_E64s_v3", "Standard_E64-16s_v3", "Standard_E64-32s_v3",
+
+                      "Standard_F64s_v2", "Standard_F72s_v2", "Standard_M128s", "Standard_M128ms", "Standard_L8s_v2",
+
+                      "Standard_L16s_v2", "Standard_L32s_v2", "Standard_L64s_v2", "Standard_L96s_v2", "SQLGL",
+
+                      "SQLGLCore", "Standard_D4_v3", "Standard_D4s_v3", "Standard_D2_v2", "Standard_DS2_v2",
+
+                      "Standard_E4_v3", "Standard_E4s_v3", "Standard_F2", "Standard_F2s", "Standard_F4s_v2",
+
+                      "Standard_D11_v2", "Standard_DS11_v2", "AZAP_Performance_ComputeV17C"};
+
+        static readonly string[] FourCoreSizesForAccelNet = { "Standard_D3_v2", "Standard_D3_v2_Promo", "Standard_D3_v2_ABC", "Standard_DS3_v2",
+
+                           "Standard_DS3_v2_Promo", "Standard_D12_v2", "Standard_D12_v2_Promo", "Standard_D12_v2_ABC",
+
+                           "Standard_DS12_v2", "Standard_DS12_v2_Promo", "Standard_F8s_v2", "Standard_F4",
+
+                           "Standard_F4_ABC", "Standard_F4s", "Standard_E8_v3", "Standard_E8s_v3", "Standard_D8_v3",
+
+                           "Standard_D8s_v3"};
+
+        sealed class ImageTypeForAccelNet
+        {
+            public string Publisher { get; }
+            public string Offer { get; }
+            public string SkuPattern { get; }
+
+            public ImageTypeForAccelNet(string publisher, string offer, string skuPattern)
+            {
+                Publisher = publisher;
+                Offer = offer;
+                SkuPattern = skuPattern;
+            }
+
+            public bool Matches(string publisher, string offer, string sku)
+            {
+                return (Publisher.Equals(publisher) && (offer == null || offer.Equals(Offer)) && (sku == null || SkuPattern == null || (new Regex(SkuPattern)).IsMatch(sku)));
```

Comment:
@praries880 do we need to do a null-check on `publisher` and/or `Publisher` here? It seems we do so for the other parameters

Reply:
We don't need to, we validate that the publisher is not null at the top of the function.

Reply:
@cormacpayne @praries880 we need Nullable Reference Types (from C# 8.0) https://blogs.msdn.microsoft.com/dotnet/2017/11/15/nullable-reference-types-in-csharp/